### ISSUE TITLE
Line endings of RStudio Project (.RProj) files

### ISSUE THREAD

**Issue by jimhester** [rstudio/rstudio#1929]
RStudio seems to always use the system default line endings (`\r\n` on Windows) when writing .RProj files, which causes diffs when contributors are working on OSs with different default line endings (or when developing in a VM). e.g. Windows and macOS.

It would be nice if saving this file took into account the line ending conversion setting like source files do.

**Comment by kevinushey** [comment]
Doesn't `git` normally convert line endings from `\r\n` to `\n`? (through the `core.autocrlf` option)

We generally write newlines as native for the platform (unless a different kind is required for some reason) and then depend on other tools to normalize line endings as appropriate in other contexts.

**Comment by jimhester** [comment]
Yes, I have this set on a few repos, but it is honestly annoying to have to do and I haven't taken the time to set it globally.

**Comment by Jakob2014uafmx** [comment]
Just to add: This problem does not only affect new project files, but also existing ones not edited by the user. For example, when creating an R project on Windows and manually converting the .RProj file to LF line endings after creation, the line endings are still reverted to CRLF each time you open the project (even if you do not change the project file at all).

**Comment by zbjornson** [comment]
> Doesn't git normally convert line endings from \r\n to \n? (through the core.autocrlf option)

A lot of people consider it best-practice to disable this and have the IDE/file-writer use line endings specified in a config file (*git is source control, it shouldn't be modifying my files*). I think adding a parameter to the `.Rproj` file like `LineEndings: <\n|\r\n|\r>` would be reasonable given that you already have similar options for spaces/tabs and encoding.

(https://support.rstudio.com/hc/en-us/community/posts/200659343-LF-line-endings)

**Comment by krlmlr** [comment]
Could the RStudio IDE use whatever line endings it finds in the existing `*.Rproj` file?

**Comment by olegat** [comment]
Not all developers use Git, and not all VCS have a substitute for `core.autocrlf`.

Even so, there are many cases where developers may keep Git's `core.autocrlf` off at all times. For example: I use Cygwin and "Ubuntu for Windows" for some things, and some POSIX tools such as `diff` and `patch` do not handle CRLF well. All text editors such as Notepad++, Visual Studio, Emacs can detect LF line-endings seamlessly, so it's not an issue at all to use LF on Windows and even preferable. The most prominent exception is Notepad.exe, but this legacy editor is rarely used in the modern age.

@krlmlr 's suggestion of detecting the EOL format of existing `*.Rproj` files is the correct approach; because this behavior is in parity with all other text editors.

Alternatively, if RStudio would use the EOL format from **Tools > Global Options > Code > Saving** then that could be a suitable solution too. In RStudio version 1.2.5033, the IDE ignores this global option for `*.Rproj` files.

**Comment by gvwilson** [comment]
Just tripped over this with IDE v1.3.959 on Windows 10.0.18363 - `git clone` in a WSL Bash shell, open project in IDE, all line endings in the `.Rproj` file are rewritten.

**Comment by kevinushey** [comment]
Marking at 1.5 so we can consider for next release.

**Comment by ismirsehregal** [comment]
This still forces me to rightclick, revert several times a day 😅...

**Comment by kevinushey** [comment]
Moving to Product Backlog milestone due to user interest.

**Comment by pcantalupo** [comment]
Any updates on this? I'm using 2023.12.1 Build 402 on Win10.  Every time I open a Rproj file it changes my line endings from \n to \r\n which causes git to detect a change. Please add a configuration option to RStudio to set the line endings that RStudio uses.  Thank you

**Comment by olivroy** [comment]
Would love to see this happening! This seems like one of the last missing pieces for happy vcs cross-platform workflow (r-lib/testthat#1958 + #14796)

It could just be written using the .Rproj line ending preference?

**Comment by DanChaltiel** [comment]
Isn't this issue already resolved?
I'm using RStudio 2024.04.2+764 "Chocolate Cosmos" on Win10 and I have the following option that seems to work perfectly 😄 
![image](https://github.com/user-attachments/assets/26df1e0c-1407-43af-b1cf-29b01c436d13)
